WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverViewController.swift

```diff
@@ -2,71 +2,55 @@ import UIKit
 import SwiftUI
 import WordPressUI
 
-final class CompliancePopoverViewController: UIViewController {
+final class CompliancePopoverViewController: UIHostingController<CompliancePopover> {
 
     // MARK: - Dependencies
 
     private let viewModel: CompliancePopoverViewModel
-
-    // MARK: - Views
-    private let hostingController: UIHostingController<CompliancePopover>
-
-    private var contentView: UIView {
-        return hostingController.view
-    }
-
     private var bannerIntrinsicHeight: CGFloat = 0
 
     init(viewModel: CompliancePopoverViewModel) {
         self.viewModel = viewModel
-        hostingController = UIHostingController(rootView: CompliancePopover(viewModel: self.viewModel))
-        super.init(nibName: nil, bundle: nil)
+        super.init(rootView: CompliancePopover(viewModel: viewModel))
     }
 
     required dynamic init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    // MARK: - View Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.addContentView()
-        hostingController.view.translatesAutoresizingMaskIntoConstraints = true
-        hostingController.rootView.goToSettingsAction = {
+        rootView.goToSettingsAction = {
             self.viewModel.didTapSettings()
         }
-        hostingController.rootView.saveAction = {
+        rootView.saveAction = {
             self.viewModel.didTapSave()
         }
         viewModel.didDisplayPopover()
     }
 
     override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
-        let targetSize = CGSize(width: view.bounds.width, height: 0)
-        let contentViewSize = contentView.systemLayoutSizeFitting(targetSize)
-        self.contentView.frame = .init(origin: .zero, size: contentViewSize)
-        self.preferredContentSize = contentView.bounds.size
-
-        self.hostingController.rootView.screenHeight = self.view.frame.height
-        self.hostingController.rootView.shouldScroll = (contentViewSize.height + 100) > self.view.frame.height
-    }
 
-    private func addContentView() {
-        self.hostingController.willMove(toParent: self)
-        self.addChild(hostingController)
-        self.view.addSubview(contentView)
-        self.hostingController.didMove(toParent: self)
+        // Make the banner scrollable when the banner height is bigger than the screen height.
+        // Send it in the next run loop to avoid a recursive `viewDidLayoutSubviews`.
+        if bannerIntrinsicHeight == 0 {
+            bannerIntrinsicHeight = view.intrinsicContentSize.height + Length.Padding.small
+            DispatchQueue.main.async {
+                self.rootView.shouldScroll = self.bannerIntrinsicHeight > self.view.frame.height
+            }
```

Comment:
This seems like a code-smell. I think SwiftUI property shouldn't be updated directly like that. Ideally, the `shouldScroll` decision should be the responsibility of `CompliancePopover` view itself and not the controller. I tried to find alternatives but nothing worked.

Anyhow, nothing to change here, but I just wanted to drop this comment. We can revisit this later.

Reply:
I don't like it as well. Main reason I put it here is that I took it from WooCommerce. And the fact that BottomSheet is a very glitchy component, and we are replicating the behavior, I did not want to spend more time on the sheet. I agree though that it looks odd.